PYME/LMVis/Extras/kinetics_recipes.py

```diff
@@ -0,0 +1,127 @@
+from PYME.recipes.base import register_module, ModuleBase, Filter, OutputModule
+from PYME.recipes.traits import Input, Output, Float, Enum, CStr, Bool, Int, List, DictStrStr, DictStrList, ListFloat, ListStr, Button, ToolbarButton
+import matplotlib as plt
+import wx
+import numpy as np
+import pylab
+from PIL import Image
+from PYME.IO import tabular
+
+from PYME.recipes.tablefilters import FilterTable
+from PYME.recipes.base import ModuleCollection
+
+from PYME.recipes import kinetics_recipes as kin_rcps
+from PYME.recipes.localisations import DBSCANClustering as dbscan
+from PYME.recipes.output import HDFOutput
+
+class DyeKinetics(object):
+    def __init__(self, visFr):
+        self.visFr = visFr
+
+        visFr.AddMenuItem('Dye Kinetics', 'Find on and off time durations for molecules', self.dye_kinetics, helpText='')
+
+    def dye_kinetics(self, event):
+        import sys
+        from time import time
+
+        t0 = time()
+
+        self.pipeline = self.visFr.pipeline
+
+        recipe = self.visFr.pipeline.recipe
+
+        script_init = kin_rcps.DyeKineticsInit(recipe, inputName=self.pipeline.selectedDataSourceKey, outputName='localisations',
+                                             clusteringRadius=40.0, blinkRadius=20.0, clusterColumnName='clusterID', fitHistograms=True,
+                                             blinkGapTolerance=0, onTimesColumnName='on_times', offTimesColumnName='off_times',
+                                             minimumClusterSize=2, minimumKeptBlinkSize=2, minimumOffTimeInSecondsFit=0,
+                                             maximumOffTimeInSecondsToFit=10000.0, offTimeFitBinWidthInSeconds=1)
+
+        if script_init.configure_traits(kind='modal'):
+
+            recipe.trait_set(execute_on_invalidation=False)
+
+            recipe.add_module(script_init)
+
+            recipe.add_module(kin_rcps.FindClumps_DK(recipe, inputName='localisations', outputName='with_clumps',
+                                                   timeWindow=script_init.blinkGapTolerance, clumpRadiusVariable=u'1.0',
```

Comment:
unicode bit here is interesting. Is that necessary?

Reply:
Yeah, since the module can take both integer multiples or 'error_x'